### DISCUSSION TITLE
Is there a way to send any RPC request without typesafety?

### DISCUSSION THREAD

**Discussion by johnpmitsch** [wevm/viem#890] [Question] [answered]
Hi all, 

Is there a way to just send a raw RPC request with a Client instance without extending the client?

i.e.
```typescript
import { createClient, createPublicClient, http } from "viem";

const client = createClient({
  transport: http("someurl"),
})

client.request({
  method: "custom_method",
  params: [...]
})
```

Right now typescript will error:

```
The expected type comes from property 'method' which is declared here on type '{ method: "web3_clientVersion"; params?: undefined; } | { method: "web3_sha3"; params: [data: `0x${string}`]; } | { method: "net_listening"; params?: undefined; } | { method: "net_peerCount"; params?: undefined; } | ... 48 more ... | { ...; }'
```

I know the client can be extended, and this is easy with `fetch` or any http lib, but this is for an SDK design so would be nice to use the client instance to unblock some requests right now without having users set up other tools. Also, we could rely on the same configuration for the client when sending the RPC request.

If it's not there, something like a `.rawRequest` might be helpful 🙏🏻

**Comment by jxom** [comment, marked as answer]
You can do `.request<RpcSchemaOverride>(…)` if you want to lax the type-safety. It is recommended to cast it to something else rather than `RpcSchemaOverride` though. The type takes form of a [`RpcSchemaOverride`](https://github.com/wagmi-dev/viem/blob/main/src/types/eip1193.ts#L1256). Example here: https://github.com/wagmi-dev/viem/blob/main/src/types/eip1193.ts#L1093 

Example:

```ts
type Override = { Parameters: Hex[], ReturnType: Hex }
const response = await client.request<Override>({ method: ‘eth_foo’, params: [‘0x…’] })
```

**Reply by johnpmitsch** [reply]
Thank you!
